### ISSUE TITLE
Error on Apply when creating a WAF rule group and attaching to WebACL

### ISSUE THREAD

**Issue by erikpaasonen** [terraform-providers/terraform-provider-aws#4078]
### Terraform Version
Terraform v0.11.5
+ provider.aws v1.13.0

### Affected Resource(s)
- aws_waf_web_acl
- aws_wafregional_web_acl

### Terraform Configuration Files
```hcl
resource "aws_waf_web_acl" "waf_acl" {
  name        = "tfWebACL"
  metric_name = "tfWebACL"

  default_action {
    type = "ALLOW"
  }

  rules {
    action {
      type = "BLOCK"
    }

    priority = 1
    rule_id  = "${aws_waf_rule_group.example.id}"
    type     = "REGULAR"
  }
}

resource "aws_wafregional_web_acl" "waf_acl" {
  name        = "tfWebACL"
  metric_name = "tfWebACL"

  default_action {
    type = "ALLOW"
  }

  rule {
    action {
      type = "BLOCK"
    }

    priority = 1
    rule_id  = "${aws_wafregional_rule_group.example.id}"
  }
}

resource "aws_waf_rule" "example" {
  name        = "example"
  metric_name = "example"
}

resource "aws_waf_rule_group" "example" {
  name        = "example"
  metric_name = "example"

  activated_rule {
    action {
      type = "COUNT"
    }

    priority = 50
    rule_id  = "${aws_waf_rule.example.id}"
  }
}

resource "aws_wafregional_rule" "example" {
  name        = "example"
  metric_name = "example"
}

resource "aws_wafregional_rule_group" "example" {
  name        = "example_group"
  metric_name = "example"

  activated_rule {
    action {
      type = "COUNT"
    }

    priority = 50
    rule_id  = "${aws_wafregional_rule.example.id}"
  }
}
```

### Debug Output
https://gist.github.com/erikpaasonen/035ca50b9ba05f74cc814a28884b608d

### Panic Output
N/A

### Expected Behavior
Terraform should attach the Rule Group object to the WebACL object in both Global and Regional contexts.

### Actual Behavior
Error message:
``` HCL
2 error(s) occurred:

* aws_waf_web_acl.waf_acl: 1 error(s) occurred:

* aws_waf_web_acl.waf_acl: Error Updating WAF ACL: Error Updating WAF ACL: WAFNonexistentItemException: The referenced item does not exist.
	status code: 400, request id: bb46863e-38fe-11e8-8e90-559f01c77452
* aws_wafregional_web_acl.waf_acl: 1 error(s) occurred:

* aws_wafregional_web_acl.waf_acl: Error Updating WAF Regional ACL: Error Updating WAF Regional ACL: WAFNonexistentItemException: The referenced item does not exist.
	status code: 400, request id: bc85e4b1-38fe-11e8-8077-2d567cf5aa85
```
Checking the WebACL in the AWS Console shows the rule set is empty, i.e. the rule group was not actually attached to the WebACL.

### Steps to Reproduce
1. `terraform apply` with the above code

### Important Factoids
Support for rule groups was very recently added to the AWS Provider. Fairly certain that this is simply an oversight during initial implementation.

### References
- #4052 
- #3898

**Comment by erikpaasonen** [comment]
Should also mention, in the case of `waf`, subsequent Apply runs succeed, but in the case of `waf-regional` it will always output this error message (no clean Apply).

**Comment by exNewbie** [comment]
I experience the same issues but on different WAF resources
`
resource "aws_waf_sql_injection_match_set" "WAFSqlInjectionDetection" {
  name = "WAFSqlInjectionDetection"

  sql_injection_match_tuples {
    text_transformation = "URL_DECODE"

    field_to_match {
      type = "QUERY_STRING"
    }
  }
}

resource "aws_wafregional_rule" "WAFSqlInjectionRule" {
  name        = "WAFSqlInjectionRule"
  metric_name = "SecurityAutomationsSqlInjectionRule"

  predicate {
    type    = "IPMatch"
    data_id = "${aws_waf_sql_injection_match_set.WAFSqlInjectionDetection.id}"
    negated = false
  }
}
`

`
[tf-waf-web-acl]$ terraform apply -auto-approve
aws_waf_sql_injection_match_set.WAFSqlInjectionDetection: Refreshing state... (ID: 896ad33c-0d67-434a-9764-d714c461838e)
aws_wafregional_rule.WAFSqlInjectionRule: Refreshing state... (ID: 3f5b593c-ff60-4279-95d9-7aa8b2bc7cd3)
aws_wafregional_rule.WAFSqlInjectionRule: Modifying... (ID: 3f5b593c-ff60-4279-95d9-7aa8b2bc7cd3)
  predicate.#:                  "0" => "1"
  predicate.4089339089.data_id: "" => "896ad33c-0d67-434a-9764-d714c461838e"
  predicate.4089339089.negated: "" => "false"
  predicate.4089339089.type:    "" => "IPMatch"

Error: Error applying plan:

1 error(s) occurred:

* aws_wafregional_rule.WAFSqlInjectionRule: 1 error(s) occurred:

* aws_wafregional_rule.WAFSqlInjectionRule: Error Updating WAF Rule: Error Updating WAF Rule: WAFNonexistentItemException: The referenced item does not exist.
        status code: 400, request id: 357fc7ac-3add-11e8-ad01-99f0b5d93ebb
`

**Comment by exNewbie** [comment]
@erikpaasonen I think I found the root cause. WAF resources suppose to be created in a specific region (in my case "ap-southeast-2"). However, they are created in *Global (Cloudfront)***. 

![Global (Cloudfront)](https://user-images.githubusercontent.com/14102056/38463063-74ffa35c-3b34-11e8-84b9-77a343dd8bac.png)

**Comment by Releng-InventiveDesigners** [comment]
This is because you are mixing aws_waf_web_acl with regional waf resources. You should use aws_wafregional_web_acl instead of aws_waf_web_acl

**Comment by jkrol2** [comment]
Reproduced with `Terraform v0.11.7`, not mixing aws_waf_web_acl  with regional waf resources. 
```hcl
resource "aws_wafregional_rate_based_rule" "example_rule" {
  name = "example_rule"
  metric_name = "example_rule"
  rate_key = "IP"
  rate_limit = 2000 

}
resource "aws_wafregional_web_acl" "example_acl" {
  depends_on = ["aws_wafregional_rate_based_rule.example_rule"]
  name = "exampleACL"
  metric_name = "exampleACL"
  default_action {
    type = "ALLOW"
  }
  rule {
    action {
      type = "BLOCK"
    }
    priority = 1
    rule_id = "${aws_wafregional_rate_based_rule.example_rule.id}"
  }
}
```

**Comment by umeshruk** [comment]
I was able to replicate the error even after using aws_wafregional_* for the rule, rule group and webacl.

**Comment by millenc** [comment]
Same issue here. I'm using only the aws_wafregional_* variants of rules, match sets, ACLs and associations. Applying multiple times does not help (at least on my tests). I'm using the following versions:

Terraform v0.11.7
+ provider.aws v1.17.0
+ provider.null v1.0.0
+ provider.random v1.2.0
+ provider.template v1.0.0

**Comment by jordimolesblanco** [comment]
Same here, using only aws_wafregional_*, not mixing.

Versions used:
Terraform: v0.11.7
provider.aws: v1.25.0, v1.24.0, v1.23.0 and v1.22.0 (didn't try earlier versions since people seem to be complaining about this anyway)

**Comment by bflad** [comment]
This should be fixed with #5053 which was just merged into master and will release with version 1.27.0 of the AWS provider, likely middle of this week. 👍 

```hcl
# Implementation is similar for aws_wafregional_web_acl resource
resource "aws_waf_web_acl" "example" {
 # ... other configuration ...
  rules {
    # ... other configuration ...
    override_action {
       type = "NONE"
    }
    type = "GROUP"
    rule_id = "${aws_waf_rule_group.example.id}"
  }
}
```

**Comment by bflad** [comment]
This has been released in [version 1.27.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1270-july-11-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
